pkg/capacity/capacity.go

```diff
@@ -562,14 +636,18 @@ func (c *Controller) syncCapacity(ctx context.Context, item workItem) error {
 		// would race with receiving that object through the event handler. In the unlikely
 		// scenario that we end up creating two objects for the same work item, the second
 		// one will be recognized as duplicate and get deleted again once we receive it.
-	} else if capacity.Capacity.Value() == quantity.Value() {
-		klog.V(5).Infof("Capacity Controller: no need to update %s for %+v, same capacity %v", capacity.Name, item, quantity)
+	} else if capacity.Capacity.Value() == quantity.Value() &&
+		(c.owner == nil || c.isOwnedByUs(capacity)) {
```

Comment:
I see, and `c.owner == nil` covers the case that a driver stays owner-less. Sgtm!